### ISSUE TITLE
index out of bounds: the len is 70141 but the index is 70141

### ISSUE THREAD

**Issue by jeffparsons** [servo/servo#13421]
index out of bounds: the len is 70141 but the index is 70141
### Servo Version:
Servo 0.0.1-d00639c
### Backtrace:
```

WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(2) }>: stop_parsing not implemented, full speed ahead!
WARNING: <Constellation>: No frame associated with pipeline PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(3) }
WARNING: <Constellation>: No frame ass
...
```
This report was generated by the browser.html issue reporter.

**Comment by jeffparsons** [comment]
Saw this clicking through from https://blog.servo.org/2016/09/19/twis-78/ from https://blog.servo.org/ a couple of minutes ago.

I may have been scrolling at the time. No idea if that's relevant.

**Comment by jdm** [comment]
I tried to reproduce this but ran into issues with outdated OpenSSL. If you happen across this same error on another site, that would be useful.

**Comment by JanZerebecki** [comment]
I could reproduce this by just waiting for a while (more than a minute, during which the page becomes incrementally more styled). No interaction required. 
```
$ ./mach run -d https://blog.servo.org/2016/09/19/twis-78/
index out of bounds: the len is 135452 but the index is 135452 (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }, at ../src/libcore/slice.rs:524)
stack backtrace:
   0:     0x561c775cc9c1 - backtrace::backtrace::libunwind::trace
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/backtrace/mod.rs:82
                         - backtrace::backtrace::trace<closure>
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/backtrace/mod.rs:70
   1:     0x561c775cd640 - backtrace::capture::{{impl}}::new
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/lib.rs:96
   2:     0x561c733eabfb - servo::main::{{closure}}
                        at /home/jan/servo/components/servo/main.rs:121
   3:     0x561c77f9d6b5 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
   4:     0x561c77f9d582 - std::panicking::begin_panic::hfbeda5aad583dc32
   5:     0x561c77f9d4c0 - std::panicking::begin_panic_fmt::h4fe9fb9d5109c4bf
   6:     0x561c77f9d441 - rust_begin_unwind
   7:     0x561c77fd9f0c - core::panicking::panic_fmt::h4395919ece15c671
   8:     0x561c77fd9eb3 - core::panicking::panic_bounds_check::he139831fe8feb00a
   9:     0x561c77c22d23 - core::slice::{{impl}}::index<u8>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/slice.rs:524
  10:     0x561c77c2bbe0 - mp3_metadata::metadata::get_id3
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/mp3-metadata-0.2.1/src/metadata.rs:36
  11:     0x561c77c3041f - mp3_metadata::metadata::read_from_slice
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/mp3-metadata-0.2.1/src/metadata.rs:365
  12:     0x561c77c09d5c - audio_video_metadata::metadata::check_mp3
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/audio-video-metadata-0.1.2/src/metadata.rs:92
  13:     0x561c77c0a305 - audio_video_metadata::metadata::get_format_from_slice
                        at /home/jan/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/audio-video-metadata-0.1.2/src/metadata.rs:124
  14:     0x561c74ffb9a9 - script::dom::htmlmediaelement::{{impl}}::check_metadata
                        at /home/jan/servo/components/script/dom/htmlmediaelement.rs:163
  15:     0x561c74ffadb0 - script::dom::htmlmediaelement::{{impl}}::data_available
                        at /home/jan/servo/components/script/dom/htmlmediaelement.rs:91
  16:     0x561c74c7e477 - net_traits::{{impl}}::process<script::dom::htmlmediaelement::HTMLMediaElementContext>
                        at /home/jan/servo/components/net_traits/lib.rs:288
  17:     0x561c750f7e25 - script::network_listener::{{impl}}::handler<net_traits::ResponseAction,script::dom::htmlmediaelement::HTMLMediaElementContext>
                        at /home/jan/servo/components/script/network_listener.rs:71
  18:     0x561c751066fc - script::script_thread::{{impl}}::handler<script::network_listener::ListenerRunnable<net_traits::ResponseAction, script::dom::htmlmediaelement::HTMLMediaElementContext>>
                        at /home/jan/servo/components/script/script_thread.rs:202
  19:     0x561c7510fefc - script::script_thread::{{impl}}::handle_msg_from_script
                        at /home/jan/servo/components/script/script_thread.rs:951
  20:     0x561c7557455c - script::script_thread::{{impl}}::handle_msgs::{{closure}}
                        at /home/jan/servo/components/script/script_thread.rs:785
  21:     0x561c7510e096 - script::script_thread::{{impl}}::profile_event<closure,core::option::Option<bool>>
                        at /home/jan/servo/components/script/script_thread.rs:879
  22:     0x561c7510c1e0 - script::script_thread::{{impl}}::handle_msgs
                        at /home/jan/servo/components/script/script_thread.rs:777
  23:     0x561c7510a7cc - script::script_thread::{{impl}}::start
                        at /home/jan/servo/components/script/script_thread.rs:631
  24:     0x561c75573284 - script::script_thread::{{impl}}::create::{{closure}}::{{closure}}
                        at /home/jan/servo/components/script/script_thread.rs:477
  25:     0x561c741f14f9 - profile_traits::mem::{{impl}}::run_with_memory_reporting<closure,fn(profile_traits::mem::ReportsChan) -> script::script_runtime::CommonScriptMsg,script::script_runtime::CommonScriptMsg,std::sync::mpsc::Sender<script::script_thread::MainThreadScriptMsg>>
                        at /home/jan/servo/components/profile_traits/mem.rs:59
  26:     0x561c758ac6b6 - script::script_thread::{{impl}}::create::{{closure}}
                        at /home/jan/servo/components/script/script_thread.rs:476
  27:     0x561c74de1baa - std::panic::{{impl}}::call_once<(),closure>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panic.rs:255
  28:     0x561c747587dc - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:356
  29:     0x561c77fa6036 - __rust_maybe_catch_panic
  30:     0x561c74585964 - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:332
  31:     0x561c7443335f - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panic.rs:311
  32:     0x561c75592401 - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/thread/mod.rs:277
  33:     0x561c74b0a519 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/liballoc/boxed.rs:591
  34:     0x561c77f9bcf0 - std::sys::thread::Thread::new::thread_start::h022e3887023c6290
  35:     0x7fe2e1944443 - start_thread
  36:     0x7fe2e146d20c - clone
  37:                0x0 - <unknown>
ERROR:servo: index out of bounds: the len is 135452 but the index is 135452
```

**Comment by jdm** [comment]
Thanks for the steps to reproduce! I am unable to do so because my OpenSSL is too old to load servo.org's certificates, unfortunately.

**Comment by jdm** [comment]
However, the backtrace shows that this is an error in @est31's mp3-metadata lib.

**Comment by jdm** [comment]
Sorry, mp3-metadata is the work of @GuillaumeGomez instead.

**Comment by JanZerebecki** [comment]
The video triggers it:
```
<!DOCTYPE html>
<html>
<body>
<video src="https://blog.servo.org/images/transform-transition-many.webm" controls="" width="70%" height="70%"></video>
</body>
</html>
```
The backtrace references this line:
https://github.com/GuillaumeGomez/mp3-metadata/blob/4c0304eab9d01caaddbe8d9c5cad57fa9ef3db36/src/metadata.rs#L36

**Comment by GuillaumeGomez** [comment]
Ah my bad. Working on a fix!

**Comment by KiChjang** [comment]
Should be fixed by #13444.
